arrow-libs/core/arrow-core/src/main/kotlin/arrow/core/Sequence.kt

```diff
@@ -659,6 +662,9 @@ fun <E, A, B> Sequence<A>.traverseEither(f: (A) -> Either<E, B>): Either<E, Sequ
   return acc.asSequence().right()
 }
 
+fun <A, B> Sequence<A>.traverseOption(f: (A) -> Option<B>): Option<Sequence<B>> =
+  traverseEither { f(it).toEither { Unit } }.orNone()
```

Comment:
```suggestion
fun <A, B> Sequence<A>.traverseOption(f: (A) -> Option<B>): Option<Sequence<B>> {
  // Note: Using a mutable list here avoids the stackoverflows one can accidentally create when using
  //  Sequence.plus instead. But we don't convert the sequence to a list beforehand to avoid
  //  forcing too much of the sequence to be evaluated.
  val acc = mutableListOf<B>()
  forEach { a ->
    when (val res = f(a)) {
      is Some -> acc.add(res.value)
      None -> return@traverseOption res
    }
  }
  return acc.asSequence().some()
}
```

Nit: a direct implementation reads better IMO, and is (slightly) more performant.
That'll save 1 allocation per element in the Sequence.